src/games/supported/DemonAttack.cpp

```diff
@@ -121,3 +129,41 @@ void DemonAttackSettings::loadState(Deserializer & ser) {
   m_lives = ser.getInt();
 }
 
+// returns a list of mode that the game can be played in
+ModeVect DemonAttackSettings::getAvailableModes(){
+    ModeVect modes;
+    modes.push_back(1);
+    modes.push_back(3);
+    modes.push_back(5);
+    modes.push_back(7);
+
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void DemonAttackSettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment){
+    if(m==1 || m==3 || m==5 || m==7){
```

Comment:
OSX doesn't support c++11's Thread Local Storage (TLS). The status of
support for various c++11 features can be seen at
http://clang.llvm.org/cxx_status.html

On Tue, Mar 29, 2016 at 5:00 PM, Marlos C. Machado <notifications@github.com
> wrote:

> In src/games/supported/DemonAttack.cpp
> <https://github.com/mgbellemare/Arcade-Learning-Environment/pull/115#discussion_r57808433>
> :
>
> > @@ -121,3 +129,41 @@ void DemonAttackSettings::loadState(Deserializer & ser) {
> >    m_lives = ser.getInt();
> >  }
> >
> > +// returns a list of mode that the game can be played in
> > +ModeVect DemonAttackSettings::getAvailableModes(){
> > +    ModeVect modes;
> > +    modes.push_back(1);
> > +    modes.push_back(3);
> > +    modes.push_back(5);
> > +    modes.push_back(7);
> > +
> > +    return modes;
> > +}
> > +
> > +// set the mode of the game
> > +// the given mode must be one returned by the previous function
> > +void DemonAttackSettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment){
> > +    if(m==1 || m==3 || m==5 || m==7){
>
> @mgbellemare <https://github.com/mgbellemare> I agree, we should move to
> C++11 and I can't think of a reason not to. However, as you said, we should
> do it in a different PR, I think.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mgbellemare/Arcade-Learning-Environment/pull/115/files/3e0f85d22e9df239aa4e17374b524efcfe59aa37#r57808433>
>